src/settings.plugin.js

```diff
@@ -257,6 +247,118 @@ wikiaJWPlayerSettingsPlugin.prototype.updateCurrentQuality = function (data) {
+		// tracks always include "off" item
+		if (event.tracks.length > 1) {
+			this.currentlySelectedCaptions = this.getSuitableCaptionsIndex(
+				this.captionLangMap[this.getUserLang()],
+				event.tracks
+			);
+
+			this.getLabelElement(this.captionsToggle)
+				.addEventListener('click', clickHandler);
+
+			this.wikiaSettingsElement.classList.remove('are-captions-empty');
+			this.player.setCurrentCaptions(this.currentlySelectedCaptions);
+			this.show();
+		} else {
+			this.getLabelElement(this.captionsToggle)
+				.removeEventListener('click', clickHandler);
+
+			this.wikiaSettingsElement.classList.add('are-captions-empty');
+		}
+	}.bind(this));
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.createCaptionsButton = function () {
+	this.captionsToggle = createToggle({
+			id: this.player.getContainer().id + '-videoCaptionsToggle',
+			label: 'Captions',
+			checked: this.config.showCaptionsToggle
+		});
+
+	this.captionsToggle.classList.add('wikia-jw-settings__captions-button');
+
+	return this.captionsToggle;
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.captionsClickHandler = function () {
+	if (this.areCaptionsOff(this.player.getCurrentCaptions())) {
+		this.player.setCurrentCaptions(this.currentlySelectedCaptions)
+	} else {
+		// "off" caption track is always the first one
+		this.player.setCurrentCaptions(0);
+	}
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.areCaptionsOff = function(captionIndex) {
+	// "off" always has first position in tracks array
+	return captionIndex === 0;
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.getUserLang = function() {
+	return window.navigator.language.slice(0, 2);
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.getSuitableCaptionsIndex = function(userLang, captionTracks) {
+	// use findIndex when ES6+ will be available
+	return captionTracks
+		.map(function (track) {
+			return track.label;
+		})
+		.indexOf(userLang);
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.captionLangMap = {
+	en: 'English',
+	pl: 'Polski',
+	fr: 'Français',
+	de: 'Deutsch',
+	it: 'Italiano',
+	ja: '日本語',
+	pt: 'Português',
+	ru: 'Русский язык',
+	es: 'Español',
+	zh: '中文'
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.getLabelElement = function (wrapper) {
+	return wrapper.querySelector('label');
+};
+
+wikiaJWPlayerSettingsPlugin.prototype.getSettingsButtonElement = function () {
+	return this.player.getContainer().querySelector('[button=' + this.buttonID + ']');
+};
+
 wikiaJWPlayerSettingsPlugin.register = function () {
 	jwplayer().registerPlugin('wikiaSettings', '8.0.0', wikiaJWPlayerSettingsPlugin);
 };
+
+function createToggle(params) {
+	var toggleWrapper = document.createElement('li'),
+		toggleInput = document.createElement('input'),
+		toggleLabel = document.createElement('label');
+
+	toggleWrapper.className = 'wikia-jw-settings__toggle';
+
+	toggleInput.setAttribute('type', 'checkbox');
+	toggleInput.setAttribute('id', params.id);
+	toggleInput.classList.add('wds-toggle__input');
```

Comment:
I think we could leverage className here as well :)

Reply:
i forgot about those little one, thx